src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -119,12 +119,6 @@ public void Flush()
         /// </summary>
         public void Dispose()
         {
-            if (this.snapshotTimer != null)
-            {
-                this.snapshotTimer.Dispose();
-                this.snapshotTimer = null;
-            }
-
             this.Flush();
```

Comment:
I think leaving disposable on `MetricManager `is an abuse of the pattern... It was designed to free consumed resources, not to remind implementor to do something... I'd like to have `IDisposable `implementation removed from `MetricManager`. I get your point that it is not a pre-req for this change, but we must not forget... Removing it later will be a breaking change...